lib/kubernetes-deploy/ejson_secret_provisioner.rb

```diff
@@ -0,0 +1,153 @@
+# frozen_string_literal: true
+require 'json'
+require 'base64'
+require 'open3'
+require 'kubernetes-deploy/kubeclient_builder'
+require 'kubernetes-deploy/logger'
+
+module KubernetesDeploy
+  class EjsonSecretError < FatalDeploymentError
+    def initialize(msg)
+      super("Creation of Kubernetes secrets from ejson failed: #{msg}")
+    end
+  end
+
+  class EjsonSecretProvisioner
+    include KubernetesDeploy::KubeclientBuilder
+
+    MANAGEMENT_ANNOTATION = "kubernetes-deploy.shopify.io/ejson-secret".freeze
+    MANAGED_SECRET_EJSON_KEY = "kubernetes_secrets".freeze
+    EJSON_SECRETS_FILE = "secrets.kubernetes-deploy.ejson".freeze
```

Comment:
I agree that the `kubernetes-deploy` feels a bit superfluous. It's not really an environment either. I think `secrets.ejson` would be fine, sort of fitting with the rest of the files instead of yaml it's ejson.